File: r/tests/test-example.R
test_that("add returns the sum of two numbers", {
  expect_equal(add(2, 3), 5)
})

test_that("normalize_name formats values correctly", {
  expect_equal(normalize_name("alice"), "Alice")
  expect_equal(normalize_name(" bob "), "Bob")
})

test_that("normalize_name rejects empty strings", {
  expect_error(normalize_name(""), "Name cannot be empty")
})

File: python/tests/test_example.py
import pytest

from example import add
from example import normalize_name


@pytest.mark.smoke
def test_add_two_numbers() -> None:
    assert add(2, 3) == 5


@pytest.mark.parametrize(
    "raw,expected",
    [
        ("alice", "Alice"),
        (" bob ", "Bob"),
    ],
)
def test_normalize_name_formats_values(raw: str, expected: str) -> None:
    assert normalize_name(raw) == expected


def test_normalize_name_rejects_empty_values() -> None:
    with pytest.raises(ValueError, match="name cannot be empty"):
        normalize_name("   ")
